src/runtime/utils.ts

```diff
@@ -516,4 +517,17 @@ export function extendBaseUrl<Context extends NuxtApp = NuxtApp>(
   }
 }
 
+export const getLocalePaths = (locale: LocaleOption): string[] => {
```

Comment:
Yeah, this is one of those duplicate codes as your mention...

Nuxt has [the concept of runtime conext and build context](https://nuxt.com/docs/guide/going-further/internals#runtime-context-vs-build-context). we want to be able to use it in both contexts.

One way to solve this, I think have a separate package for use within the nuxt i18n monorepo workspace.

I wish there was a better way to do more code sharing in context... 

@danielroe 
Do you have best solution?

Reply:
I managed to remove the duplicate code by preparing the options during generation instead of at runtime